qcodes/tests/dataset/test_database_creation_and_upgrading.py

```diff
@@ -151,6 +155,76 @@ def test_perform_actual_upgrade_1_to_2():
         assert len(c.fetchall()) == 2
 
 
+def test_perform_actual_upgrade_2_to_3():
+
+    v2fixpath = os.path.join(fixturepath, 'db_files', 'version2')
+
+    if not os.path.exists(v2fixpath):
+        pytest.skip("No db-file fixtures found. You can generate test db-files"
+                    " using the scripts in the legacy_DB_generation folder")
+
+    dbname_old = os.path.join(v2fixpath, 'empty.db')
+
+    with temporarily_copied_DB(dbname_old, debug=False, version=2) as conn:
+
+        assert get_user_version(conn) == 2
+
+        desc_query = 'SELECT run_description FROM runs'
+
+        with pytest.raises(RuntimeError):
+            atomic_transaction(conn, desc_query)
+
+        perform_db_upgrade_2_to_3(conn)
+
+        assert get_user_version(conn) == 3
+
+        c = atomic_transaction(conn, desc_query)
+        assert len(c.fetchall()) == 0
+
+    dbname_old = os.path.join(v2fixpath, 'some_runs.db')
+
+    with temporarily_copied_DB(dbname_old, debug=False, version=2) as conn:
+
+        assert get_user_version(conn) == 2
+
+        perform_db_upgrade_2_to_3(conn)
+
+        desc_query = 'SELECT run_description FROM runs'
+
+        c = atomic_transaction(conn, desc_query)
+        assert len(c.fetchall()) == 10
+
+        # retrieve the yaml string and recreate the object
+
+        sql = f"""
+              SELECT run_description
+              FROM runs
+              WHERE run_id == 1
+              """
+        c = atomic_transaction(conn, sql)
+        yaml_str = one(c, 'run_description')
+
+        desc = RunDescriber.from_yaml(yaml_str)
+        idp = desc.interdeps
+        assert isinstance(idp, InterDependencies)
+
+        # here we verify that the dependencies encoded in
+        # tests/dataset/legacy_DB_generation/generate_version_2.py
+        # are recovered
+
+        p1 = [p for p in idp.paramspecs if p.name == 'p0'][0]
```

Comment:
it's a bit weird that the instance of a parameter with name "p0" is called "p1" :) the same with p5 and p4.

Reply:
Fixed.

Reply:
( for some reason, i don't see the fix; i reloaded the github page several times )